### PR TITLE
TNO-2023, TNO-2024: Modify / view searches

### PR DESCRIPTION
**Pull request by jdtoombs**
- users may now select the gear icon and click modify search to modify viewed search
- common tooltip component added called TooltipMenu to be used when a menu is to be displayed with clickable options
- few UI adjustments
- cut out a number of requests being triggered erroneously 

modify-search

### CODE REVIEW

File: app/subscriber/src/components/sidebar/advanced-search/AdvancedSearch.tsx [hunk 1]
```diff
@@ -43,13 +54,17 @@ export const AdvancedSearch: React.FC<IAdvancedSearchProps> = ({
   onSearchPage,
 }) => {
   const navigate = useNavigate();
```
Comment (jdtoombs) [inline app/subscriber/src/components/sidebar/advanced-search/AdvancedSearch.tsx:56]: rearranged to be easier to read, can't remember if this is the same order you do @Fosol?  or if I got it backwards :P

Comment (jdtoombs) [reply app/subscriber/src/components/sidebar/advanced-search/AdvancedSearch.tsx:56]: customhooks

reacthooks

File: app/subscriber/src/components/tooltip-menu/TooltipMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
```
Comment (jdtoombs) [inline app/subscriber/src/components/tooltip-menu/TooltipMenu.tsx:1]: Consistent them for toolbars that act as a menu of items

File: app/subscriber/src/css/_variables.scss [hunk 3]
```diff
@@ -1,45 +1,37 @@
 // The purpose of this file is to help us create a theme so that when UI colour changes we can control them in one place.
```
Comment (jdtoombs) [inline app/subscriber/src/css/_variables.scss:1]: formatter went off on this...

File: app/subscriber/src/components/sidebar/advanced-search/AdvancedSearch.tsx (hunk 1 above)
Comment (Fosol) [reply app/subscriber/src/components/sidebar/advanced-search/AdvancedSearch.tsx:56]: I always sort by hook dependencies first, then state, then variables, then functions.

**Review by Fosol — APPROVED** [review]

**Pull request merged**